packages/types/src/interfaces/event-bus.ts

```diff
@@ -0,0 +1,27 @@
+import { TransactionBaseService } from "@medusajs/medusa"
```

Comment:
That is an excellent suggestion!

Reply:
No avoid relying on the core, I would have to move the abstract `TransactionBaseService` into `@medusajs/utils`. I am not sure, I like the idea of extending the scope of this PR that much. What's you take?

Reply:
No you wouldn't have to do that, we are extending the `TransactionBaseService` just to use it as an interface, but if we create the interface, you just need to extends this interface instead of the class directly. It was the lazy approach. Then the `TransactionBaseService` in the core can just implements this new interface